docs/src/integration_test_tool.md

```diff
@@ -1,3 +1,161 @@
+
+With the package filter `r "*"`, which means that all dependencies are traversed, the result would be `["MyPkgMain", "ExternDep1"]` if we search for `MyPkgA`. We are not the maintainer of `ExternDep1`, so we can hardly change code in `ExternDep1` and it makes little sense to test `ExternDep1` with our code changes.
+
+## Package Filter: Special Cases
+
+### Dependencies of non matching packages
+
+In the following diagram, `MyPkgA` is not found with the package filter `r"^MyPkg*"`. The reason for this is that the traversal algorithm does not traverse `ExternDep1` and therefore does not check the dependencies of `ExternDep1`.
+
+```mermaid
+graph TD
+    MyPkgMain -->|using| ExternDep1 -->|using| MyPkgA
+```
+
+### The top graph package
+
+Sometimes the top graph package does not have the same naming scheme as all other packages. The section [Project Dependency Tree](#Project-Dependency-Tree) contains examples of this. If the top package is not included in the package filter, the entire graph is not traversed and the result is empty every time.
+
+```mermaid
+graph TD
+    MetaTestPkg -->|using| MyPkgMain -->|using| MyPkgA
+    MyPkgMain -->|using| MyPkgB
+```
+
+`depending_projects()` returns `[]` for the package filter `r"^MyPkg*"` and the package `MyPkgB`. The package filter `r"^MyPkg*|^MetaTestPkg$` returns `["MyPkgA"]`.
+
+# Project Dependency Tree
+
+We strongly recommend using a `project.toml` of a package from the package ecosystem as input for `depending_projects()` instead of defining the project dependency graph manually. The reason for this is that a manually defined dependency tree can easily become obsolete. In contrast, the `Project.toml` must be up to date, otherwise the Julia application will not work.
+
+The crucial question is which `Project.toml` should be used. The dependency graph of a `Project.toml` can only be constructed in one direction. You can only use the dependencies of the packages. From which package a package is used can only be determined if you came from the package before. Let's assume we have the following package ecosystem:
+
+```mermaid
+graph TD
+    MyPkgMain -->|using| MyPkgA
+    MyPkgMain -->|using| MyPkgB -->|using| MyPkgD
+    MyPkgB -->|using| MyPkgE -->|using| MyPkgF
+    MyPkgMain -->|using| MyPkgC
+```
+
+If we take the `Project.toml`[\*] of `MyPkgMain`, we get the complete graph. If we take the `Project.toml` of `MyPkgB`, we get the following graph:
+
+```mermaid
+graph TD
+    MyPkgB -->|using| MyPkgD
+    MyPkgB -->|using| MyPkgE -->|using| MyPkgF
+```
+
+!!! note
+
+    The dependency tree that is created with the `Project.toml` of `MyPkgMain` does not contain `MyPkgMain` itself. This is a technical detail from Julia. It will contain the dependencies `MyPkgA` to `MyPkgF`. To solve the problem, an additional package is needed. The additional package only has the dependency `MyPkgMain`:
```

Comment:
```suggestion
    The dependency tree created with the `Project.toml` of `MyPkgMain` does not contain `MyPkgMain` itself. This is a technical detail from Julia. It will contain the dependencies `MyPkgA` to `MyPkgF`. To solve the problem, an additional package is needed, which has only the dependency `MyPkgMain`:
```